azure-spring-boot/src/test/java/com/microsoft/azure/spring/cloundfoundry/environment/AzureCloudFoundryServiceApplicationTest.java

```diff
@@ -113,4 +113,43 @@ public void testVcapSingleServiceWithNulls() {
         }
     }
 
+    @Test
+    @SuppressFBWarnings("DM_DEFAULT_ENCODING")
+    public void testVcapUserProvidedService() {
+        final Resource resource = new ClassPathResource("/vcap3.json");
+        final String content;
+        try {
+            content = new String(
+                    Files.readAllBytes(Paths.get(resource.getURI())));
+            final VcapResult result = parser.parse(content);
+            final VcapPojo[] pojos = result.getPojos();
+            assertNotNull(pojos);
+            assertEquals(1, pojos.length);
+            final VcapPojo pojo = pojos[0];
+
+            LOG.debug("pojo = " + pojo);
+            assertEquals(4, pojo.getCredentials().size());
+            assertEquals(0, pojo.getTags().length);
+            assertEquals(0, pojo.getVolumeMounts().length);
+            assertEquals("user-provided", pojo.getLabel());
+            assertNull(pojo.getProvider());
+            assertEquals("azure-documentdb", pojo.getServiceBrokerName());
+            assertEquals("mydocumentdb", pojo.getServiceInstanceName());
+            assertEquals("standard", pojo.getServicePlan());
+            assertNull(pojo.getSyslogDrainUrl());
+
+            assertEquals("docdb123mj",
+                    pojo.getCredentials().get("documentdb_database_id"));
+            assertEquals("dbs/ZFxCAA==/",
+                    pojo.getCredentials().get("documentdb_database_link"));
+            assertEquals("https://hostname:443/",
+                    pojo.getCredentials().get("documentdb_host_endpoint"));
+            assertEquals(
+                    "3becR7JFnWamMvGwWYWWTV4WpeNhN8tOzJ74yjAxPKDpx65q2lYz60jt8WXU6HrIKrAIwhs0Hglf0123456789==",
+                    pojo.getCredentials().get("documentdb_master_key"));
+        } catch (IOException e) {
+            LOG.error("Error reading json file", e);
```

Comment:
@mindprogenitor  If IOException is encountered, the test will exit silently, throw the Exception out and make the test fail ? Thanks.

Reply:
Never mind, will open another PR to fix this.

Reply:
Think I blindly copied the other test case, but you are absolutely right, should have seen it.